pkg/cloudprovider/providers/openstack/openstack.go

```diff
@@ -196,6 +197,12 @@ type AuthOpts struct {
 	Region           string `name:"os-region"`
 	CAFile           string `gcfg:"ca-file" mapstructure:"ca-file" name:"os-certAuthorityPath" value:"optional"`
 
+	EndpointType gophercloud.Availability `gcfg:"endpoint-type" mapstructure:"endpoint-type" name:"os-endpointType" value:"optional"`
```

Comment:
Should be ideally interface_type or interface ?

Reply:
I thought about using a regular string, but it will require type conversion inside every package. Therefore I decided to keep the `gophercloud.Availability`.